source/routing/raptor_solution_reader.cpp

```diff
@@ -575,21 +409,60 @@ std::vector<Path> read_solutions(const RAPTOR& raptor,
             } catch (stop_search&) {}
         }
     }
-    std::vector<Path> sol;
-    for (const auto& s: reader.solutions) {
-        sol.push_back(make_path(s, reader));
-    }
-    return sol;
+    return reader.solutions;
 }
 
 } // anonymous namespace
 
-std::vector<Path> read_solutions(const RAPTOR& raptor,
-                      const bool clockwise,
-                      const RAPTOR::vec_stop_point_duration& deps,
-                      const RAPTOR::vec_stop_point_duration& arrs,
-                      const bool disruption_active,
-                      const type::AccessibiliteParams& accessibilite_params)
+bool Journey::better_on_dt(const Journey& that, bool request_clockwise) const {
+    if (request_clockwise) {
+        if (arrival_dt != that.arrival_dt) { return arrival_dt <= that.arrival_dt; }
+        if (departure_dt != that.departure_dt) { return departure_dt >= that.departure_dt; }
+    } else {
+        if (departure_dt != that.departure_dt) { return departure_dt >= that.departure_dt; }
+        if (arrival_dt != that.arrival_dt) { return arrival_dt <= that.arrival_dt; }
+    }
+    // FIXME: I don't like this objective, for me, this is a
+    // transfer objective, but then you can return some solutions
+    // that we didn't return before.
+    if (sections.size() != that.sections.size()) { return true; }
```

Comment:
No, I want to do the comparisons after only if the 2 solutions have the same number of sections. But, as said in the comment, this comparison function sucks.